### ISSUE TITLE
Segfault on TX2 with cufft

### ISSUE THREAD

**Issue by wentasah** [CTU-IIG/kcf#4]
I'm getting segfault on TX2 (see GDB session below). Could the reason be that I'm using OpenCV 2.4.9? README says, I should use version 3.0. I don't understand why I can compile it but it fails at runtime.
```console
sojka@tegra-ubuntu:~/kcf/build-cufft$ gdb --args ./kcf_vot -v -f ~/dataset/bag/
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
...
(gdb) r
Starting program: /home/sojka/kcf/build-cufft/kcf_vot -v -f /home/sojka/dataset/bag/
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/aarch64-linux-gnu/libthread_db.so.1".
[New Thread 0x7faaa2ee50 (LWP 16628)]
resizing image horizontaly by factor of 0.341333 and verticaly by factor of 0.368345
init: img size 480 360
init: win size. 128 128
init: min max scales factors: 0.158556 3.09179
FFT: cuFFT

Thread 1 "kcf_vot" received signal SIGSEGV, Segmentation fault.
0x0000007fb7b8b7fc in ?? () from /usr/lib/aarch64-linux-gnu/libopencv_core.so.2.4
(gdb) bt
#0  0x0000007fb7b8b7fc in ?? () from /usr/lib/aarch64-linux-gnu/libopencv_core.so.2.4
#1  0x0000007fb7b9e7f4 in ?? () from /usr/lib/aarch64-linux-gnu/libopencv_core.so.2.4
#2  0x0000007fb7b9f414 in cv::multiply(cv::_InputArray const&, cv::_InputArray const&, cv::_OutputArray const&, double, int) () from /usr/lib/aarch64-linux-gnu/libopencv_core.so.2.4
#3  0x0000007fb7c22d28 in cv::MatOp_Bin::assign(cv::MatExpr const&, cv::Mat&, int) const ()
   from /usr/lib/aarch64-linux-gnu/libopencv_core.so.2.4
#4  0x00000000004302cc in cv::Mat::operator= (this=0x7fffffe338, e=...)
    at /usr/include/opencv2/core/mat.hpp:1419
#5  0x000000000043af74 in cuFFT::forward_window (this=0x6e8180, 
    input=std::vector of length 44, capacity 62 = {...}) at ../src/fft_cufft.cpp:208
#6  0x0000000000425e60 in KCF_Tracker::init (this=0x7fffffeb68, img=..., bbox=..., fit_size_x=128, 
    fit_size_y=128) at ../src/kcf.cpp:180
#7  0x00000000004221c4 in main (argc=4, argv=0x7ffffff4d8) at ../main_vot.cpp:105

```
This is on current master branch.

**Comment by Shanigen** [comment]
I tried to run all CUDA version of the tracker on the TX2 with OpenCV 2.4.9 and it didn't fail at runtime even once. As for the confusion, why you can compile the tracker with OpenCV 2.4.9 even though the README says, that it requires OpenCV 3.0+ That is something that was true only in the beginning of the development of the CUDA versions of the tracker, but now all versions works with OpenCV 2.4+. I will correct the README.

**Comment by wentasah** [comment]
Note that I've updated the README a lot in master.

**Comment by wentasah** [comment]
This was probably caused by missing error checking and not having CUDA driver installed. I fixed this by commit 334c883ce9864ab31752416a0aae3ab244ddee7f.
